### PR TITLE
:sparkles: (feat): Add Support for Hardened Images

### PR DESCRIPTION
**Pull request by KrisJohnstone**
**Description of changes:**
When using hardened images (i.e. CIS or STIG) there are a few changes required to get everything running and have nodes join the cluster.

**Testing Done**
Here's a hardened image running in one of our test clusters:
image
image

Note: Accidentally closed old PR.

### CODE REVIEW

**Comment by Issacwww** [comment]
Could you please share some context?
why only AL2023 need this change?

**Comment by KrisJohnstone** [comment]
> Could you please share some context? why only AL2023 need this change?

AL2 EOL is just under a year away. For anyone whom going forth wants stateramp/fedramp accreditation, using AL2 as a basis is a bad idea (IMO). And from what issues I'd found during research it seemed like support was already there in AL2 (I've not tested personally).

File: templates/al2023/provisioners/hardened-images.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+set -o pipefail
+set -o nounset
+set -o errexit
+
+if [[ "$HARDENED_IMAGE" == "true" ]]; then
```
Comment (bryantbiggs) [inline templates/al2023/provisioners/hardened-images.sh:7]: what are the downsides of setting these by default? where would this pose an issue?

**Comment by cartermckinnon** [comment]
"Hardened" is very subjective 😄 and I don't think we want to attempt defining it in this repository at this time -- that will be a much larger effort. We'll continue tracking it in #245

**Comment by KrisJohnstone** [comment]
@cartermckinnon what about if i changed the var to be SELINUX_ENFORCED or something like that?

File: templates/al2023/provisioners/hardened-images.sh (hunk 1 above)
Comment (KrisJohnstone) [reply templates/al2023/provisioners/hardened-images.sh:7]: Sorry my backlogs been rather large recently and missed this. Providing the binary exists by default then I can't foresee any.

**Pull request closed without merging**